papiea-engine/__tests__/docs_tests/api_docs.test.ts

```
@@ -293,6 +293,57 @@ describe("API docs test entity", () => {
         });
     });
 
+    test("Provider with procedures generates correct openAPI removing all variables with 'x-papiea' - 'status_only' from properties and required of a spec", async () => {
+        expect.hasAssertions()
+        const provider = new ProviderBuilder("provider_include_all_props").withVersion("0.1.0").withKinds([getSpecOnlyKind()]).build()
+        const providerDbMock = new Provider_DB_Mock(provider)
+        const apiDocsGenerator = new ApiDocsGenerator(providerDbMock)
+        const kind_name = provider.kinds[0].name
+        const structure = provider.kinds[0].kind_structure[kind_name]
+
+        // add 'z' to required field, so that we check it gets removed from required and fields
+        structure.required.push("z")
+
+        // add required fields to v that are marked as status-only so that we check recursive deletion works
+        structure.properties.v.required = ["h"]
+        structure.properties.v.properties["h"] = {"type": "number", "x-papiea": "status-only"}
+
+        // set x to spec-only to see that the implementation doesn't remove it from the required fields
+        structure.properties.x["x-papiea"] = "spec-only"
```

There is no `spec-only` field attribute, only a `spec-only` entity. We should probably throw an error on this.

---

Also, I think we need to add a check that we don't have any `status-only` fields on `spec-only` entities

---

I've added 2 additional steps during validation of provider:
1) check that the value of x-papiea is ["status-only'] (as these are all possible values for now)
2) check that if an entity is a spec-only entity, x-papiea can be only one of [""] (as these are all possible values except status-only)